YDotNet/Document/Transactions/Transaction.cs

```diff
@@ -20,33 +19,34 @@ namespace YDotNet.Document.Transactions;
 ///         All operations that need to touch or modify the contents of a document need to be executed through a
 ///         transaction.
 ///     </para>
-///     <para>A <see cref="Transaction" /> is automatically committed during <see cref="Dispose" />.</para>
+///     <para>A <see cref="Transaction" /> is automatically committed during <see cref="IDisposable.Dispose" />.</para>
 /// </remarks>
-public class Transaction : IDisposable
+public class Transaction : UnmanagedResource
 {
-    /// <summary>
-    ///     Initializes a new instance of the <see cref="Transaction" /> class.
-    /// </summary>
-    /// <param name="handle">The handle to the native resource.</param>
-    internal Transaction(nint handle)
+    private readonly Doc doc;
+
+    internal Transaction(nint handle, Doc doc)
+        : base(handle)
     {
-        Handle = handle;
+        this.doc = doc;
+
+        doc.NotifyTransactionStarted();
```

Comment:
If you have a better idea, I am open for it. It is a workaround until it is fixed in rust.

Reply:
I don't have any ideas right now, I'll just leave it documented in a `TODO` for future reference.

It's acceptable for me that we're throwing a managed exception (because the returned `Transaction` pointer is zero since there are other transactions open).